workflow-core/src/commonMain/kotlin/com/squareup/workflow1/StatefulWorkflow.kt

```diff
@@ -73,7 +75,377 @@ public abstract class StatefulWorkflow<
 
   public inner class RenderContext internal constructor(
     baseContext: BaseRenderContext<PropsT, StateT, OutputT>
-  ) : BaseRenderContext<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT> by baseContext
+  ) : BaseRenderContext<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT> by baseContext {
+
+    /**
+     * Like [eventHandler], but no-ops if [state][WorkflowAction.Updater.state] has
+     * changed to a different type than [CurrentStateT] by the time [update] fires.
+     *
+     *    when(renderState) {
+     *      is NewGame -> {
+     *        NewGameScreen(
+     *          onCancel = context.safeEventHandler<NewGame> {
+     *            setOutput(CanceledStart)
+     *          },
+     *          onStartGame =
+     *            context.safeEventHandler<NewGame, String, String> { currentState, x, o ->
+     *              state = Playing(currentState.gameType, PlayerInfo(x, o))
+     *            }
+     *        )
+     *      }
+     *
+     * This is not an uncommon case. Consider accidental rapid taps on
```

Comment:
Do you think it would be helpful to mention an event handler does not synchronously execute its update lambda? I find a lot of people don't understand that it creates and queues a WorkflowAction to be executed later. (At least I think that's what happens. Hopefully I have that correct!) Understanding that can help make it clearer why there's time for a second button tap to queue a second WorkflowAction before the first WorkflowAction is executed and changes state/produces a new rendering without the event handler, and then those WorkflowActions are pulled off the queue and executed in sequence.

Reply:
That's a really good callout Jeff, I'll add that.